tests/unit-tests.js

```diff
@@ -0,0 +1,158 @@
+
+ */
+"use strict";
+var fluid  = require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var jqUnit = require("node-jqunit");
+
+fluid.require("%gpii-testem/src/js/testem-component.js");
+
+fluid.defaults("gpii.tests.testem.eventTestingHarness", {
+    gradeNames: ["fluid.component"],
+    events: {
+        myEvent: null
+    },
+    invokers: {
+        getWrappedPromise: {
+            funcName: "gpii.testem.generateSingleUseEventListener",
+            args:     ["{that}", "{that}.events.myEvent"]
+        }
+    }
+});
+
+jqUnit.module("Testing event wrapper....");
+jqUnit.test("A promise should resolve when the associated event is fired...", function () {
+    jqUnit.expect(2);
+    jqUnit.stop();
+    var testComponent = gpii.tests.testem.eventTestingHarness();
+    var eventPromise = testComponent.getWrappedPromise();
+    eventPromise.then(
+        function () {
+            jqUnit.start();
+            jqUnit.assert("The promise should have been resolved.");
+        },
+        function () {
+            jqUnit.start();
+            fluid.fail("The promise should not have been rejected.");
+        }
+    );
+
+    testComponent.events.myEvent.fire();
+
+    // Fire the event again to confirm that no attempt is made to resolve the promise again.
+    testComponent.events.myEvent.fire();
+
+    jqUnit.assert("It should be safe to fire the event again...");
+});
+
+jqUnit.test("The event listener should not throw an error if the promise is resolved elsewhere...", function () {
+    jqUnit.expect(1);
+    var testComponent = gpii.tests.testem.eventTestingHarness();
+    var eventPromise = testComponent.getWrappedPromise();
+    eventPromise.resolve();
+
+    // Fire the event and confirm that no attempt is made to resolve the promise again.
+    testComponent.events.myEvent.fire();
+
+    jqUnit.assert("It should be safe to fire the event once the promise is resolved...");
+});
+
+jqUnit.test("The event listener should not throw an error if the promise is rejected elsewhere...", function () {
+    jqUnit.expect(1);
+    var testComponent = gpii.tests.testem.eventTestingHarness();
+    var eventPromise = testComponent.getWrappedPromise();
+    eventPromise.reject();
+
+    // Fire the event and confirm that no attempt is made to resolve the promise again.
+    testComponent.events.myEvent.fire();
+
+    jqUnit.assert("It should be safe to fire the event once the promise is rejected...");
+});
+
+
+jqUnit.module("Testing promise timeout wrapper...");
+
+jqUnit.test("A promise should reject with a timeout if it is not rejected/resolve externally...", function () {
+    jqUnit.expect(1);
+
+    jqUnit.stop();
+    var timeoutPromise = gpii.testem.addPromiseTimeout(fluid.promise(), "Timed out as expected...", 250);
+
+    timeoutPromise.then(
+        function () {
+            jqUnit.start();
+            fluid.fail("The wrapped promise should not have been resolved.")
```

Comment:
When the linting is fixed, I imagine we should also exclude gpii-testem\tests\testem-fixtures\instrumented from linting